contrib/pg_tde/src/catalog/tde_keyring.c

```diff
@@ -916,6 +920,33 @@ load_kmip_keyring_provider_options(char *keyring_options)
 	return kmip_keyring;
 }
 
+#define MAX_FILE_DATA_LENGTH 1024
+
+static char *
+get_file_value(const char *path, const char *field_name)
+{
+	int			fd = -1;
+	char	   *val;
+
+	fd = BasicOpenFile(path, O_RDONLY);
+	if (fd < 0)
+	{
+		elog(ERROR, "failed to open file \"%s\" for \"%s\"", path, field_name);
+	}
+
+	val = palloc0(MAX_FILE_DATA_LENGTH);
+	if (pg_pread(fd, val, MAX_FILE_DATA_LENGTH, 0) == -1)
+	{
+		close(fd);
+		elog(ERROR, "failed to read file \"%s\" for \"%s\"", path, field_name);
```

Comment:
Shouldn't we look at the errno for both the message and the error code here? And shouldn't it be an ereport call?